fix(document): use <section>s with aria-labelledby

caugner opened the pull request:
## Summary

Makes sure we use `<section>` elements with ~~`aria-describedby`~~ `aria-labelledby` in MDN articles.

### Problem

Previously, we were just using `<div>`s to wrap MDN article sections, which is not ideal.

### Solution

I replaced the `<div>`s with `<section>`s and added ~~`aria-describedby`~~ `aria-labelledby` where possible.

---

## Screenshots

_No visual changes._

---

## How did you test this change?

I did not see any differences, **but** I did not check for potential CSS rules targeting those `div`s.

caugner commented on client/src/document/ingredients/prose.tsx:
```diff
@@ -1,10 +1,21 @@
 }
 
-export function ProseWithHeading({ id, section }) {
+export function ProseWithHeading({
```
I wonder whether we couldn't just merge `Prose` and `ProseWithHeading` now that both accept the id parameter. 🤔

schalkneethling replied:
I would be open to that.

caugner replied:
Done in 1ba27e1.

schalkneethling commented:
> Talked about this with @fiji-flo last week, and what we probably want is to have the sections contain the heading. Then we would only have sections within an article, and the section would contain the heading next to the other items (paragraphs, tables).

What I was really getting at is, we might not need `section` at all. Without using either `aria-label` or `aria-labelledby` a `section` element is pretty much the same as using a `div`. See this article for reference: https://www.smashingmagazine.com/2020/01/html5-article-section/#so-what-about-lt-section-gt

schalkneethling commented:
> > Talked about this with @fiji-flo last week, and what we probably want is to have the sections contain the heading. Then we would only have sections within an article, and the section would contain the heading next to the other items (paragraphs, tables).
> 
> What I was really getting at is that we might not need `section` at all. Without using either `aria-label` or `aria-labelledby` a `section` element is pretty much the same as using a `div`. See this article for reference: https://www.smashingmagazine.com/2020/01/html5-article-section/#so-what-about-lt-section-gt

Also, creating too many `regions` or landmarks on a page can actually be less helpful to screen reader users.

caugner commented:
> Without using either `aria-label` or `aria-labelledby` a `section` element is pretty much the same as using a `div`.

Oh, good catch! I added `aria-describedby` in this PR for what should have been `aria-labelled-by`.

@schalkneethling Wouldn't this improve the situation, having labelled sections exclusively inside the article?

schalkneethling commented:
> @schalkneethling Wouldn't this improve the situation, having labelled sections exclusively inside the article?

It might but as mentioned, some of these sections simply wrap a single paragraph sometimes. We would have to see how common that is. Just want to make sure we do not make it too noisy by adding all these regions to the page.

schalkneethling commented:
@caugner What would be the resulting DOM if we do not wrap these with anything? i.e. not a `div` nor a `section` element. Perhaps there are style hooks that rely on this though?

caugner commented:
@schalkneethling I have now ...

1. Merged `Prose` with `ProseWithHeading`.
2. Replaced `aria-describedby` with `aria-labelledby`.
3. Moved the heading inside the section.

The resulting `<article>` structure is as follows:

<img width="1079" alt="image" src="https://user-images.githubusercontent.com/495429/163246766-51d69238-7160-40e0-92dc-ba6ad7e3bb1f.png">

It's not perfect yet, because our document structure is not hierarchical (e.g. the two Example regions belong to the "Accessibility concerns", but are not inside that section), but I think this is an improvement over the status quo, where the content of each section is not associated with the corresponding heading.

schalkneethling reviewed (approved):
Thanks so much for this, @caugner 🎉
